sys/x86/include/db_machdep.h

```diff
@@ -0,0 +1,120 @@
+/*-
+ * Mach Operating System
+ * Copyright (c) 1991,1990 Carnegie Mellon University
+ * All Rights Reserved.
+ *
+ * Permission to use, copy, modify and distribute this software and its
+ * documentation is hereby granted, provided that both the copyright
+ * notice and this permission notice appear in all copies of the
+ * software, derivative works or modified versions, and any portions
+ * thereof, and that both notices appear in supporting documentation.
+ *
+ * CARNEGIE MELLON ALLOWS FREE USE OF THIS SOFTWARE IN ITS "AS IS"
+ * CONDITION.  CARNEGIE MELLON DISCLAIMS ANY LIABILITY OF ANY KIND FOR
+ * ANY DAMAGES WHATSOEVER RESULTING FROM THE USE OF THIS SOFTWARE.
+ *
+ * Carnegie Mellon requests users of this software to return to
+ *
+ *  Software Distribution Coordinator  or  Software.Distribution@CS.CMU.EDU
+ *  School of Computer Science
+ *  Carnegie Mellon University
+ *  Pittsburgh PA 15213-3890
+ *
+ * any improvements or extensions that they make and grant Carnegie Mellon
+ * the rights to redistribute these changes.
+ *
+ * $FreeBSD$
+ */
+
+#ifndef _X86_DB_MACHDEP_H_
+#define	_X86_DB_MACHDEP_H_
+
+#include <machine/frame.h>
+#include <machine/trap.h>
+
+typedef	vm_offset_t	db_addr_t;	/* address - unsigned */
+#ifdef __amd64__
+typedef	long		db_expr_t;	/* expression - signed */
+#else
+typedef	int		db_expr_t;	/* expression - signed */
```

Comment:
Does it break a lot if you just use long on i386 as well?

Reply:
I tested and hit two cases (`db_printf("%r",...)` and a type mismatch with `size_t`). Changing this to `register_t` might work.